src/layout/execute.cpp

```diff
@@ -60,32 +61,23 @@ int execute(chopper::configuration & config, std::vector<std::vector<std::string
     }
 
     seqan::hibf::layout::layout hibf_layout;
-    std::vector<seqan::hibf::sketch::hyperloglog> sketches;
     std::vector<size_t> kmer_counts;
-
-    seqan::hibf::concurrent_timer compute_sketches_timer{};
-    seqan::hibf::concurrent_timer union_estimation_timer{};
-    seqan::hibf::concurrent_timer rearrangement_timer{};
-    seqan::hibf::concurrent_timer dp_algorithm_timer{};
-
-    compute_sketches_timer.start();
-    seqan::hibf::sketch::compute_sketches(config.hibf_config, kmer_counts, sketches);
-    compute_sketches_timer.stop();
+    seqan::hibf::sketch::estimate_kmer_counts(sketches, kmer_counts);
 
     if (config.determine_best_tmax)
     {
         hibf_layout = determine_best_number_of_technical_bins(config, kmer_counts, sketches);
     }
     else
     {
-        dp_algorithm_timer.start();
+        config.dp_algorithm_timer.start();
         hibf_layout = seqan::hibf::layout::compute_layout(config.hibf_config,
                                                           kmer_counts,
                                                           sketches,
                                                           seqan::hibf::iota_vector(sketches.size()),
-                                                          union_estimation_timer,
-                                                          rearrangement_timer);
-        dp_algorithm_timer.stop();
+                                                          config.union_estimation_timer,
```

Comment:
Maybe we want to put the timers into the hibf config at some point?

Reply:
yes I guess